src/main/java/dev/j3fftw/litexpansion/items/PortableCharger.java

```diff
@@ -0,0 +1,225 @@
+    }
+
+    @Override
+    public ItemUseHandler getItemHandler() {
+        return e -> {
+            e.cancel();
+
+            // Get variables
+            final Player p = e.getPlayer();
+            final ItemStack chargerItem = e.getItem();
+            final Rechargeable charger = (Rechargeable) SlimefunItem.getByItem(chargerItem);
+
+            // Create GUI Items
+            Inventory inventory = Bukkit.createInventory(null, 27, ChatColor.GOLD + "Portable Charger");
+
+            ItemStack backgroundItem = buildNonInteractable(Material.GRAY_STAINED_GLASS_PANE, null);
+            ItemStack borderItem = buildNonInteractable(Material.YELLOW_STAINED_GLASS_PANE, null);
+            ItemStack powerItem = buildNonInteractable(Material.GLOWSTONE, "&4Power");
+
+            // Build and open GUI
+            for (int i = 0; i < 27; i++)
+                inventory.setItem(i, backgroundItem);
+
+            for (int slot : border)
+                inventory.setItem(slot, borderItem);
+
+            inventory.setItem(powerSlot, powerItem);
+            updateSlot(inventory, powerSlot, "&6&lPower Remaining",
+                "&e" + charger.getItemCharge(chargerItem) + "J");
+            inventory.clear(chargeSlot);
+            p.openInventory(inventory);
+
+            // Task that triggers every second
+            new BukkitRunnable(){
+                public void run() {
+
+                    ItemStack deviceItem = inventory.getItem(chargeSlot);
+                    SlimefunItem sfItem = SlimefunItem.getByItem(deviceItem);
+
+                    if (sfItem instanceof Rechargeable && !isItem(deviceItem)) {
+
+                        Rechargeable device = (Rechargeable) sfItem;
+                        float neededCharge = device.getMaxItemCharge(deviceItem)
+                            - device.getItemCharge(deviceItem);
+                        float availableCharge = charger.getItemCharge(chargerItem);
+
+                        // Three different scenarios
+                        if (p.getGameMode() == GameMode.CREATIVE && neededCharge > 0) {
+                            device.setItemCharge(deviceItem, device.getMaxItemCharge(deviceItem));
+
+                        } else if (neededCharge > 0 && availableCharge > 0) {
+
+                            if (neededCharge >= chargeSpeed && availableCharge >= chargeSpeed) {
+                                charger.removeItemCharge(chargerItem, chargeSpeed);
+                                device.addItemCharge(deviceItem, chargeSpeed);
+
+                            } else if (neededCharge < availableCharge) {
+                                charger.removeItemCharge(chargerItem, neededCharge);
+                                device.addItemCharge(deviceItem, neededCharge);
+
+                            } else {
+                                charger.removeItemCharge(chargerItem, availableCharge);
+                                device.addItemCharge(deviceItem, availableCharge);
```

Comment:
device.addItemCharge var plox
 charger.removeItemCharge var plox

Reply:
what this is a method

Reply:
Can still make it a var